ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -60,6 +76,32 @@ void drawTextInput(SkCanvas *canvas,
   // paintParagraph
   yOffset = (layout.getContentFrame().size.height - paragraph->getHeight()) / 2;
   paragraph->paint(canvas, frame.origin.x + layout.contentInsets.left, frame.origin.y + layout.contentInsets.top + yOffset);
+
+  // draw Cursor
+  position = cursor_.end - cursor_.locationFromEnd;
+  if (cursor_.locationFromEnd == cursor_.end) {
+    rects = paragraph->getRectsForRange(0, position+1, RectHeightStyle::kTight, RectWidthStyle::kTight);
+    cursorRect.fLeft =  frame.origin.x + rects[0].rect.left() + layout.contentInsets.left;
+    cursorRect.fRight = frame.origin.x + rects[0].rect.left() + layout.contentInsets.left;
+  }
+  else {
+    rects = paragraph->getRectsForRange(0, position, RectHeightStyle::kTight, RectWidthStyle::kTight);
+    cursorRect.fLeft =  frame.origin.x + rects[0].rect.right() + layout.contentInsets.left;
```

Comment:
the cursorRect value is always updated using rects[0] ie., 0th index .. But we should be using the textbox value at position index right .. unless we are getting only one textbox here .
can you confirm this behavior.